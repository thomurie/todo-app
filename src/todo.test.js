import { fireEvent, render, screen } from "@testing-library/react";
import Todo from "./todo";

// Smoke
test("renders learn react link", () => {
  render(<Todo />);
});

// Snapshot
test("Matches snapshot", () => {
  const { asFragment } = render(<Todo />);
  expect(asFragment()).toMatchSnapshot();
});

// Critical Business Logic
test("Title Renders", () => {
  render(<Todo task="Test Components" />);
  expect(screen.queryByText("Test Components")).toBeInTheDocument();
});

test("Buttons Render", () => {
  const { queryByText } = render(<Todo task="Test Components" />);
  expect(queryByText("Complete")).toBeInTheDocument();
  expect(queryByText("Delete")).toBeInTheDocument();
});
