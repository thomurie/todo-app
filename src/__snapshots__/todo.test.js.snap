// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Todo"
  >
    <div
      class="Todo-Task"
    >
      <h3
        class="Todo-title"
      />
    </div>
    <div
      class="Todo-Options"
    >
      <button>
        Complete
      </button>
      <button>
        Delete
      </button>
    </div>
  </div>
</DocumentFragment>
`;
